threading/channels.nim

```diff
@@ -286,39 +288,43 @@ proc `=copy`*[T](dest: var Chan[T], src: Chan[T]) =
   `=destroy`(dest)
   dest.d = src.d
 
-proc trySend*[T](c: Chan[T], src: sink Isolated[T]): bool {.inline.} =
+proc trySend*[T](c: Chan[T], src: var Isolated[T]): bool {.inline.} =
```

Comment:
Why was this change necessary?

Reply:
Because trySend can fail and that means whatever was sinked should be destroyed. And if trySend is used in a loop that creates copies. Maybe T cannot be copied. This PR removes all possible copies.

Reply:
But then `trySend(chan, isolate MyObject(...))` does not compile anymore. Maybe introduce `trySendMut` instead (or something with a better name).

Reply:
Done, I added trySendMut (name seems fine).

Reply:
No, the name sucks as it stresses what happens but not why it happens. The name should be `trySendNocopy` but then people (myself included) wonder why the `sink` does not ensure "no copy" already. What is really going on here?

Reply:
@Araq I tried to explain in a [comment](https://github.com/nim-lang/threading/pull/74#issuecomment-2289837169)

Or read mratim's view in [rfc#347](https://github.com/nim-lang/RFCs/issues/347#issuecomment-791986889)

Reply:
There is no problem here with `sink` or copies per se. `sink` takes over ownership, always, and ideally `trySend` takes over ownership conditionally. As such `var` works better but one needs to watch out not to reuse the object afterwards.

Maybe it should be named `tryTake`.